lib/jekyll/hooks.rb

```diff
@@ -88,19 +80,17 @@ def self.insert_hook(owner, event, priority, &block)
     end
 
     # interface for Jekyll core components to trigger hooks
-    def self.trigger(instance, event, *args)
-      owner_symbol = OWNER_MAP[instance.class]
-
+    def self.trigger(owner, event, *args)
```

Comment:
The situation that prompted this change was one where a user had subclassed Page. Let's say you do that by declaring ```class MyPage < Jekyll::Page; end```. The expected behavior is at certain stages of the build, Jekyll should trigger hooks.

The problem is Jekyll doesn't currently know that MyPage should trigger :page hooks. MyPage relies on its superclass and potentially other internal Jekyll components to trigger hooks at the right times. But if what we have is an instance of MyPage, how can we translate that into the right hash key, :page in this case, to look up the hooks.

I had a few ideas and this PR represents the simplest, but maybe also the ugliest from the caller's perspective. But here's a list of the ideas I've had so far. Let me know if you have any others:

1) Require the trigger caller to pass they key in the trigger method call (this PR)

2) Give every type of owner an instance method, class method or constant (something heritable) that calls out the owner symbol. Take this change to Jekyll::Page for example:
```ruby
module Jekyll
  class Page
    def self.hook_owner; :page; end
  end
end
```
Instances of Jekyll::Page and instances of Jekyll::Page descendants could now just ask the instance of MyPage or instance of Jekyll::Page for its .class.hook_owner and know how to find the hooks. This could essentially replace the lookup table at the top of Jekyll::Hooks and make each hook owner declare such a method.

The idea is the same, but there could be some more sugary syntax:
```ruby
module Jekyll
  class Page
    ...
  end
  Hooks.add_owner :page, Page
end
```

3) Make the Hooks class search the class hierarchy of the instances passed in. For example if you pass in an instance of MyPage (subclass of page), Hooks would quiz that instance to see if it is_a?(Jekyll::Page) and if so trigger :page hooks. If not, ask to see if it is_a?(Jekyll::Post) and so on until it has checked all the possible hook owners it could be.